### DISCUSSION TITLE
Is it possible to exclude specific list of field in Invoke-PnPSiteTemplate commandlet?

### DISCUSSION THREAD

**Discussion by mohammadamer** [pnp/powershell#2841] [Q&A]
Is it possible to exclude specific list of field in Invoke-PnPSiteTemplate commandlet?
I know that we can exclude using -ExcludeHandlers but I stumbled upon exclude specific list or field. Is that possible?

**Comment by AndersRask** [comment]
What I usually do, is to split it up in different XML files with individual lists in each, instead of having one huge XML file with everything for a site, and using Handlers to select what to pick from the template.
This gives me flexibility to invoke specific lists for one site and not for another, and also gives me flexibility to maintain the XML for individual Content Types or Lists.
I don't do this for fields, as I usually just drop all the fields into a site, but you definitely could split fields up the same way as well..

Hope this helps
Anders Rask

**Reply by mohammadamer** [reply]
Thanks for your answer, This would help as long as it seems that there is no way of -ExcludeHandlers for specific list or field.

**Reply by mohammadamer** [reply]
Hi @AndersRask 
Regarding your answer and splitting the template in different XML files. I was struggling on apply a list template that has a choice field to a site.
The issue is every time, I apply the list template to the site, the choices options that have been added by end users are gone because it's overwritten by the template that's why my initial idea was to avoid applying the list template.

Have you faced such issue where you wanted to apply list template without overwriting a choice field options.

**Reply by AndersRask** [reply]
That is a very specific case, so no I have not.
But if you have end users that modify the list and fields, there will always be a chance that you overwrite what they have done, depending on where they are modifying.
One way to have more control would be to prohibit modifying in lists on active sites, and in stead let them modify a "template" site, that you regularily append on the sites already created.
Else you will have to either manually or programmatically clean up the template xml files for fields that you dont want to overwrite

hth
